### PR TITLE
feature/61921 load testing scenarios 2024

### PR DESCRIPTION
**Pull request by GuyHarwood**
update load testing scenarios & scripts for 2024

- adapt JMX live scenario to the new check submission API
- create Dockerfile for flood agent 
- create Util APIs for a 'taken check'
- update documentation

### CODE REVIEW

File: deploy/build-agent/build-push-docker-hub.sh [hunk 1]
```diff
@@ -9,5 +9,5 @@ if [ -n "$GIT_STATUS_OUTPUT" ]; then
 fi
 
 GIT_COMMIT_SHORT=$(git log -1 --format=%h)
-DOCKER_BUILDKIT=1 docker build -t stamtc/devops-build-agent:latest -t stamtc/devops-build-agent:$GIT_COMMIT_SHORT --platform linux/amd64 .
```
Comment (jon-shipley) [inline deploy/build-agent/build-push-docker-hub.sh:12]: I take it that DOCKER_BUILDKIT=1 is no longer needed.  Can you explain why please?  Did the standard docker build task get better?

File: load-test/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FROM alpine:3.19
+# FROM ubuntu:22.04
```
Comment (jon-shipley) [inline load-test/Dockerfile:2]: Looks like cruft that can be deleted

File: load-test/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+RUN apk add --no-cache curl
+# RUN apt-get update && apt-get install -y curl
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
+RUN curl -L https://github.com/flood-io/flood-agent/releases/download/v1.0/flood-agent-v1.0-linux-64bit > flood-agent
+RUN chmod +x flood-agent
+COPY ./docker-start.sh .
+RUN chmod +x docker-start.sh
+
+CMD ["/bin/sh", "-c", "./flood-agent --token ${FLOOD_TOKEN} --grid ${FLOOD_GRID}"]
```
Comment (jon-shipley) [inline load-test/Dockerfile:16]: Nice 😄

File: load-test/docker-start.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+set -e
+
+./flood-agent --token "$FLOOD_TOKEN" --grid "$FLOOD_GRID"
```
Comment (jon-shipley) [inline load-test/docker-start.sh:4]: This file does not start docker?  Is it still required?

**Review by jon-shipley — APPROVED** [review]
💯 LGTM - a couple of questions added in the review though.

File: deploy/build-agent/build-push-docker-hub.sh (hunk 1 above)
Comment (GuyHarwood) [reply deploy/build-agent/build-push-docker-hub.sh:12]: https://docs.docker.com/build/buildkit/#:~:text=Getting%20started,t%20need%20to%20enable%20BuildKit.

Pleased to report it's now a default...

`BuildKit is the default builder for users on Docker Desktop, and Docker Engine as of version 23.0.`

File: load-test/Dockerfile (hunk 3 above)
Comment (GuyHarwood) [reply load-test/Dockerfile:16]: Thanks for your help with this! 😁

File: load-test/docker-start.sh (hunk 4 above)
Comment (GuyHarwood) [reply load-test/docker-start.sh:4]: good catch, will remove

**Pull request merged**